nodes/src/nodes/memory_persistent/memory_store.py

```
@@ -0,0 +1,613 @@
+        """Apply TTL to all keys belonging to a session."""
+        if ttl_seconds is None or ttl_seconds <= 0:
+            return
+        keys_to_expire = [
+            self._keys_key(session_id),
+            self._history_key(session_id),
+            self._meta_key(session_id),
+        ]
+        # Also expire all data keys
+        members = self._client.smembers(self._keys_key(session_id))
+        for member in members:
+            keys_to_expire.append(self._data_key(session_id, member))
```

Ruff (0.15.7)

[warning] Use `list.extend` to create a transformed list

(PERF401)